datalake/backend/processing/check_metadata_content/task.py

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env python3
+import sys
+from argparse import ArgumentParser
+from json import load
+from os.path import dirname, join
+from typing import Callable, TextIO
+from urllib.parse import urlparse
+
+import boto3
+from botocore.response import StreamingBody
+from jsonschema import FormatChecker, validate
+
+SCHEMA_PATH = join(dirname(__file__), "stac-spec/collection-spec/json-schema/collection.json")
+
+
+def validate_url(url: str, url_reader: Callable[[str], TextIO]) -> None:
+    url_stream = url_reader(url)
+    url_json = load(url_stream)
+
+    with open(SCHEMA_PATH) as schema_file:
+        schema_json = load(schema_file)
+
+    validate(url_json, schema_json, format_checker=FormatChecker())
+
+
+def parse_arguments():
+    argument_parser = ArgumentParser()
+    argument_parser.add_argument("urls", nargs="+")
+    arguments = argument_parser.parse_args()
+    return arguments
+
+
+def s3_url_reader() -> Callable[[str], StreamingBody]:
+    client = boto3.client("s3")
+
+    def read(href: str) -> StreamingBody:
+        parse_result = urlparse(href, allow_fragments=False)
+        bucket_name = parse_result.netloc
+        key = parse_result.path[1:]
+        response = client.get_object(Bucket=bucket_name, Key=key)
+        return response["Body"]
+
+    return read
+
+
+def main() -> int:
+    arguments = parse_arguments()
```

Comment:
* according Detailed Design, this script should be called with following parameters
```
[
  "--dataset-id", "Ref::dataset_id",
  "--version-id", "Ref::version_id",
  "--type", "Ref::type",
  "--metadata-url", "Ref::metadata_url"
]
```

Reply:
Only the metadata URL is relevant for this job, so I don't see why we should. We might end up having to do that in some future iteration, of course.